### ISSUE TITLE
[bug]: TypeError exception when using `--save_intermediates` in CLI mode

### ISSUE THREAD

**Issue by yaph** [invoke-ai/InvokeAI#2733]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Linux

### GPU

cuda

### What happened?

I entered `"volcano" --save_intermediates 2 --steps 20` as a prompt after starting `./invokeai.sh` and chose the command line mode.

I looked into the code and tried to debug this. The `callback` function defined [here](https://github.com/invoke-ai/InvokeAI/blob/34e449213c0b65ffd64a2980a17b2445fc6b43ca/ldm/invoke/CLI.py#L1265) is only called with one argument [here](https://github.com/invoke-ai/InvokeAI/blob/e3d1c64b77548fdfb2341c2a9369c0d9aeb016c8/ldm/invoke/generator/diffusers_pipeline.py#L179).

### Additional context

```
(stable-diffusion-1.5) invoke> "volcano" --save_intermediates 2 --steps 20
>> Intermediate images will be written into /Applications/invokeai/outputs/intermediates/000328
>> patchmatch.patch_match: INFO - Compiling and loading c extensions from "/Applications/invokeai/.venv/lib/python3.10/site-packages/patchmatch".
>> patchmatch.patch_match: ERROR - patchmatch failed to load or compile (Command 'make clean && make' returned non-zero exit status 2.).
>> patchmatch.patch_match: INFO - Refer to https://invoke-ai.github.io/InvokeAI/installation/060_INSTALL_PATCHMATCH/ for installation instructions.
>> Patchmatch not loaded (nonfatal)
Generating:   0%|                                                                                           | 0/1 [00:00<?, ?it/s]
>> An error occurred:
Traceback (most recent call last):
  File "/Applications/invokeai/.venv/lib/python3.10/site-packages/ldm/invoke/CLI.py", line 164, in main
    main_loop(gen, opt)
  File "/Applications/invokeai/.venv/lib/python3.10/site-packages/ldm/invoke/CLI.py", line 394, in main_loop
    gen.prompt2image(
  File "/Applications/invokeai/.venv/lib/python3.10/site-packages/ldm/generate.py", line 516, in prompt2image
    results = generator.generate(
  File "/Applications/invokeai/.venv/lib/python3.10/site-packages/ldm/invoke/generator/base.py", line 112, in generate
    image = make_image(x_T)
  File "/Applications/invokeai/.venv/lib/python3.10/site-packages/ldm/invoke/generator/txt2img.py", line 40, in make_image
    pipeline_output = pipeline.image_from_embeddings(
  File "/Applications/invokeai/.venv/lib/python3.10/site-packages/ldm/invoke/generator/diffusers_pipeline.py", line 340, in image_from_embeddings
    result_latents, result_attention_map_saver = self.latents_from_embeddings(
  File "/Applications/invokeai/.venv/lib/python3.10/site-packages/ldm/invoke/generator/diffusers_pipeline.py", line 366, in latents_from_embeddings
    result: PipelineIntermediateState = infer_latents_from_embeddings(
  File "/Applications/invokeai/.venv/lib/python3.10/site-packages/ldm/invoke/generator/diffusers_pipeline.py", line 183, in __call__
    callback(result)
TypeError: make_step_callback.<locals>.callback() missing 1 required positional argument: 'step'
```

**Comment by lstein** [comment]
Thank you. I'll have a look.

**Comment by lstein** [comment]
This is now fixed. Thanks for the report! The fix will appear in 2.3.1

**Comment by yaph** [comment]
Thank you for the quick fix and your wonderful contributions to open source.
